### PR TITLE
Fix rhbz#1647812 and package-lock.json

### PR DESCRIPTION
**Pull request by b-ranto**
This includes the fix for rhbz#1647812 - Ceph-Mgr does not properly feed mgr's perf_counter schema (the part where ceph-mgr metadata is missing). It also contains a fix for the issue with package-lock.json file that I hit while testing the back-port/trying to build the branch.

The fix for rhbz#1647812 includes fixes for:

https://tracker.ceph.com/issues/23330
https://tracker.ceph.com/issues/23286

This results in the back-port of the following PRs (all the patches applied cleanly) :

https://github.com/ceph/ceph/pull/20866
https://github.com/ceph/ceph/pull/20875

### CODE REVIEW

File: src/mgr/DaemonServer.cc [hunk 1]
```diff
@@ -1591,24 +1591,27 @@ void DaemonServer::got_mgr_map()
...
-	  daemon_state.insert(daemon);
-	  dout(10) << "added missing " << key << dendl;
+	if (!daemon_state.exists(key) && !daemon_state.is_updating(key)) {
+	  md_update(key);
+	  dout(10) << "triggered addition of " << key << " via metadata update" << dendl;
 	}
       }
       for (auto& i : mgrmap.standbys) {
-	DaemonKey key("mgr", i.second.name);
+        DaemonKey key("mgr", i.second.name);
```
[the remark this answers is not in the document]
Comment (b-ranto) [reply src/mgr/DaemonServer.cc]: Hmm, wouldn't this cause a potential problem when cherry-picking other commits?

btw: I did not change the original commits, they all applied cleanly.

**Comment by b-ranto** [comment]
@alfonsomthd Yeah, it would be better if it was in another PR but since we are not actually merging the PRs but picking up commits that we apply on top of rhcs-3.2, I included it here for the review.

File: src/mgr/DaemonServer.cc (hunk 1 above)
[the remark this answers is not in the document]
Comment (epuertat) [reply src/mgr/DaemonServer.cc]: Ok, let's not add more noise then ;-)

**Comment by epuertat** [comment]
Ok, let's go with that then

**Pull request merged**